### PR TITLE
Fix datapath in nftests

### PR DESCRIPTION
**Pull request by asp8200**
https://github.com/nf-core/sarek/issues/1425  (Implicitly https://github.com/nf-core/sarek/issues/1406)

Also fixing `docker.runOptions` in test-configs and profile config for GATK Spark, since it doesn't have support for `docker.userEmulation`. See https://github.com/nf-core/sarek/pull/1405#issuecomment-1967034480.

All tests pass with `docker.runOptions = { params.use_gatk_spark ? '' : '-u $(id -u):$(id -g)' }.call()`. However, the following command, which is the joint-germline test-cmd plus GATK-spark, fails :
```
NXF_VER=24.01.0-edge PROFILE=docker nextflow run main.nf -profile test_cache,targeted,use_gatk_spark --input ./tests/csv/3.0/mapped_joint_bam.csv --tools haplotypecaller --step variant_calling --joint_germline --outdir results --known_snps_vqsr false --known_indels_vqsr false
```
Now there is not much point in specifying the profile `use_gatk_spark` for a command like that, since it doesn't use GATK-spark modules. However, the same cmd but with `NXF_VER=23.04.0` works on the master branch.

The pytests and the above test-cmd all work if I do

```
docker.runOptions = { params.use_gatk_spark ? '-u $(id -u):$(id -g) -v /etc/passwd:/etc/passwd:ro ' : '-u $(id -u):$(id -g)' }.call()
```
TO-DO: Remove temporary tag `foo`.

### CODE REVIEW

File: nextflow.config [hunk 1]
```diff
@@ -192,7 +192,7 @@ profiles {
         docker.enabled         = true
         podman.enabled         = false
         shifter.enabled        = false
-        docker.runOptions      = '-u $(id -u):$(id -g)'
+        docker.runOptions      = '-u $(id -u):$(id -g) -v /etc/passwd:/etc/passwd:ro -v /etc/shadow:/etc/shadow:ro -v /etc/group:/etc/group:ro'
```
Comment (maxulysse) [inline nextflow.config:195]: Happy with that as a solve in the tests, not sure about using that here.

File: conf/test.config [hunk 2]
```diff
@@ -108,6 +108,7 @@ if (System.getenv('PROFILE')) {
     } else if ("$PROFILE" == "docker") {
         conda.enabled                            = false
         docker.enabled                           = true
+        docker.runOptions                        = '-u $(id -u):$(id -g) -v /etc/passwd:/etc/passwd:ro -v /etc/shadow:/etc/shadow:ro -v /etc/group:/etc/group:ro'
```
Comment (asp8200) [inline conf/test.config:111]: Here we have `docker.userEmulation   = { params.use_gatk_spark ? false : true }.call()`  on the master-branch.

File: nextflow.config (hunk 1 above)
Comment (asp8200) [inline nextflow.config:195]: Here we have `docker.userEmulation   = { params.use_gatk_spark ? false : true }.call()`  on the master-branch.

File: conf/test.config (hunk 2 above)
Comment (FriederikeHanssen) [reply conf/test.config:111]: could consider to only overwrite when running with spark & joint germline and not generally?

File: tests/test_gatk4spark.yml [hunk 3]
```diff
@@ -3,6 +3,7 @@
   tags:
     - gatk4spark
     - preprocessing
+    - foo
```
Comment (FriederikeHanssen) [inline tests/test_gatk4spark.yml:6]:
```suggestion
```

**Review by pontus — APPROVED** [review]

File: tests/test_gatk4spark.yml (hunk 3 above)
Comment (maxulysse) [inline tests/test_gatk4spark.yml:6]: let's delete that

**Review by maxulysse — APPROVED** [review]

**Pull request merged**